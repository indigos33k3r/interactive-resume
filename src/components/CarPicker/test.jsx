import React from 'react'
import {shallow} from 'enzyme'

describe('CarPicker', () => {
  it('renders <CarPicker />', () => {
    const component = render()

    expect(component).toMatchSnapshot()
  })

  context('when <CarPickerPick /> with `prop.pickerClass` equal "atruck" was clicked', () => {
    it('calls #props.pickCarHandler', () => {
      const pickCarHandler = jest.fn()
      const component = render({pickCarHandler})
      component.find('CarPickerPick[pickerClass="atruck"]').prop('pickCarHandler')()

      expect(pickCarHandler).toHaveBeenCalledTimes(1)
    })
  })

  context('when <CarPickerPick /> with `prop.pickerClass` equal "batmobil" was clicked', () => {
    it('calls #props.pickCarHandler', () => {
      const pickCarHandler = jest.fn()
      const component = render({pickCarHandler})
      component.find('CarPickerPick[pickerClass="batmobil"]').prop('pickCarHandler')()

      expect(pickCarHandler).toHaveBeenCalledTimes(1)
    })
  })

  context('when `props.hide` is `true`', () => {
    it('renders <CarPicker /> with "car-picker--hide" class', () => {
      const component = render({hide: true})

      expect(component).toMatchSnapshot()
    })
  })

  function render(props = {}) {
    const defaultProps = {
      pickCarHandler: () => {},
      hide: false
    }
    const CarPicker = require('.').default

    return shallow(<CarPicker {...defaultProps} {...props} />)
  }
})
