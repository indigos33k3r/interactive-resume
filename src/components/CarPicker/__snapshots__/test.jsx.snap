// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarPicker renders <CarPicker /> 1`] = `
<div
  className="car-picker"
>
  <div
    className="car-picker__background"
  >
    <div
      className="car-picker__car-selection"
    >
      <h1
        className="car-picker__header"
      >
        Pick your car
        <span
          className="car-picker__header-subtext"
        >
          For the ride!
        </span>
      </h1>
      <div
        className="car-picker__box"
      >
        <CarPickerPick
          pickCarHandler={[Function]}
          pickerClass="atruck"
        />
      </div>
      <div
        className="car-picker__box"
      >
        <CarPickerPick
          pickCarHandler={[Function]}
          pickerClass="batmobil"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`CarPicker when \`props.hide\` is \`true\` renders <CarPicker /> with "car-picker--hide" class 1`] = `
<div
  className="car-picker car-picker--hide"
>
  <div
    className="car-picker__background"
  >
    <div
      className="car-picker__car-selection"
    >
      <h1
        className="car-picker__header"
      >
        Pick your car
        <span
          className="car-picker__header-subtext"
        >
          For the ride!
        </span>
      </h1>
      <div
        className="car-picker__box"
      >
        <CarPickerPick
          pickCarHandler={[Function]}
          pickerClass="atruck"
        />
      </div>
      <div
        className="car-picker__box"
      >
        <CarPickerPick
          pickCarHandler={[Function]}
          pickerClass="batmobil"
        />
      </div>
    </div>
  </div>
</div>
`;
